cylc/uiserver/data_store_mgr.py

```diff
@@ -67,40 +71,98 @@ def __init__(self, workflows_mgr):
 
-    async def sync_workflow(self, w_id, *args, **kwargs):
+    def update_contact(self, w_id, contact_data=None):
+        delta = DELTAS_MAP[WORKFLOW]()
+        delta.time = time.time()
+        flow = delta.updated
+        flow.id = w_id
+        if contact_data:
+            # update with contact file data
+            flow.name = contact_data['name']
+            flow.owner = contact_data['owner']
+            flow.host = contact_data[CFF.HOST]
+            flow.port = int(contact_data[CFF.PORT])
+            # flow.pub_port = int(contact_data[CFF.PUBLISH_PORT])
+            flow.api_version = int(contact_data[CFF.API])
+        else:
+            # wipe pre-existing contact-file data
+            flow.owner, flow.name = w_id.split(ID_DELIM)
+            flow.host = ''
+            flow.port = 0
+            # flow.pub_port = 0
+            flow.api_version = 0
+            flow.status = SuiteStatus.Stopped.value
+
+        # Apply to existing workflow data
+        apply_delta(WORKFLOW, delta, self.data[w_id])
+        if 'delta_times' not in self.data[w_id]:
+            self.data[w_id]['delta_times'] = {}
+        self.data[w_id]['delta_times'][WORKFLOW] = delta.time
+        # Queue delta for subscription push
+        self.delta_store_to_queues(w_id, ALL_DELTAS, delta)
+
+    async def sync_workflow(self, w_id, contact_data):
         """Run data store sync with workflow services.
 
         Subscriptions and sync management is instantiated and run in
         a separate thread for each workflow. This is to avoid the sync loop
         blocking the main loop.
 
         """
+        logger.debug(f'sync_workflow({w_id})')
         if self.loop is None:
             self.loop = asyncio.get_running_loop()
+
+        # don't sync if subscription exists
         if w_id in self.w_subs:
             return
 
         self.delta_queues[w_id] = {}
+        self.update_contact(w_id, contact_data)
 
         # Might be options other than threads to achieve
         # non-blocking subscriptions, but this works.
         self.executors[w_id] = ThreadPoolExecutor()
         self.executors[w_id].submit(
-            partial(self.start_subscription, w_id, *args, **kwargs)
+            partial(
+                self.start_subscription,
+                w_id,
+                contact_data['name'],
+                contact_data[CFF.HOST],
+                contact_data[CFF.PUBLISH_PORT]
+            )
         )
         await self.entire_workflow_update(ids=[w_id])
 
-    def purge_workflow(self, w_id):
+    async def register_workflow(self, w_id, name, owner):
+        logger.debug(f'register_workflow({w_id})')
+        self.delta_queues[w_id] = {}
+
+        # create new entry in the data store
+        data = deepcopy(DATA_TEMPLATE)
+        self.data[w_id] = data
+
+        # create new entry in the delta store
+        self.update_contact(w_id)
+
+    def stop_workflow(self, w_id):
+        logger.debug(f'stop_workflow({w_id})')
```

Comment:
For some reason these still get printed in normal (non-debug) `jupyterhub` run:
```
2020-09-06 10:22:32,006 cylc.uiserver.data_store_mgr DEBUG    sync_workflow(sutherlander|baz)
2020-09-06 10:23:06,974 cylc.uiserver.data_store_mgr DEBUG    purge_workflow(sutherlander|baz)
2020-09-06 10:23:07,177 cylc.uiserver.data_store_mgr DEBUG    stop_workflow(sutherlander|baz)
```
@kinow - Any ideas?

Reply:
@dwsutherland our logging configuration is separated from `jupyterhub`'s. If the user runs `jupyterhub --debug` that should affect only JupyterHub logging.

In our case, we can either provide a different logging configuration, or you can locally modify `cylc/uiserver/logging_config.json`, change the console handler to `INFO`. Then you should get only:

```
2020-09-06 11:15:25,826 cylc.uiserver.main INFO     JupyterHub Service Prefix: /user/kinow/
2020-09-06 11:15:25,827 cylc.uiserver.main INFO     Listening on 42747 and serving static content from /home/kinow/Development/python/workspace/cylc-uiserver/../cylc-ui/dist
2020-09-06 11:15:25,827 cylc.uiserver.main INFO     Starting Cylc UI
2020-09-06 11:15:25,827 cylc.uiserver.main INFO     /home/kinow/Development/python/workspace/cylc-uiserver/../cylc-ui/dist

```

Reply:
(It could be that we either need to document this well, or maybe use the same logging level from jupyterhub? No strong preference, hadn't thought about that until now :sweat_smile: )

Reply:
Ok, not this PR to solve!